spec/openhab/dsl/rules/builder_spec.rb

```diff
@@ -70,6 +70,70 @@ def triggered?
       end
     end
 
+    describe "#on_start" do
+      it "works with default level" do
+        rule = rule do
+          on_start
+          run { nil }
+        end
+        t = rule.triggers.first
+        expect(t.type_uid).to eql "core.SystemStartlevelTrigger"
+        expect(t.configuration.properties.key?("startlevel")).to be true
+        expect(t.configuration.properties["startlevel"].to_i).to eq 100
+      end
+
+      # @param [Symbol,Integer,Array<Symbol>,Array<Integer>] level
+      # @param [Integer,Array<Integer>] rawlevel
+      def test_on_start(level, raw_level = nil)
+        rule = rule do
+          if level.is_a?(Array)
+            on_start at_levels: level
+          else
+            on_start at_level: level
+          end
+          run { nil }
+        end
+
+        level = Array.wrap(level)
+        raw_level = raw_level ? Array.wrap(raw_level) : level
+
+        rule.triggers.to_a.each.with_index do |trigger, i|
```

Comment:
no need for .to_a.each.with_index. java.util.List already responds to each_with_index:

`rule.triggers.each_with_index do |trigger, i|`

Reply:
thanks, didn't know that.